src/classes/groundplaces.ts

```diff
@@ -0,0 +1,131 @@
+   * @param fromStopGroupGpuid Ground place unique identifier of the old stopGroup.
+   * @param intoStopGroupGpuid Ground place unique identifier of the new stopGroup.
+   * @returns {Void}
+   */
+  // WARNING: The segmentProviderStop cannot be without a parent.
+  public moveSegmentProviderStop(segmentProviderId: string, fromStopGroup: string, intoStopGroup: string): void {}
+
+  /**
+   * @description Move a stopGroup from a stopCluster to another stopCluster.
+   * @param stopGroupToMoveGpuid Ground place unique identifier of the stopGroup to move.
+   * @param fromStopClusterGpuid Ground place unique identifier of the old stopCluster.
+   * @param intoStopClusterGpuid Greound place unique identifier of the new stopCluster.
+   * @returns {StopCluster}
+   */
+  // @ts-ignore
+  public moveStopGroup(stopGroupToMoveGpuid: string, fromStopClusterGpuid: string, intoStopClusterGpuid: string): StopCluster {}
+
+  /**
+   * @description Delete place only if it's empty.
+   * @param placeToRemoveGpuid Ground place unique identifier of the place to remove.
+   * @returns {Void}
+   */
+  public deletePlace(placeToRemoveGpuid: string): void {}
+
+  /**
+   * @description Add a stopGroup to a stopCluster.
+   * @param stopGroupToAddGpuid Ground place unique identifier of the stopGroup to add.
+   * @param intoStopClusterGpuid Ground Place unique identifier of the stopCluster.
+   * @returns {StopCluster}
+   */
+  // @ts-ignore
+  public addStopGroupToStopCluster(stopGroupToAddGpuid: string, intoStopClusterGpuid: string): StopCluster {}
+
+  /**
+   * @description Merge two stopGroups. It means moving all segmentProviderStop of a stopGroup into another.
+   * @param stopGroupToMergeGpuid Ground place unique identifier of the stopGroup to merge.
+   * @param intoStopGroupGpuid Ground Place unique identifier of the stopGroup to be merged.
+   * @returns {StopGroup}
+   */
+  // @ts-ignore
+  // WARNING: Check first if the merged stopGroup don't have two segmentStopProvider of the same segmentProvider in it.
+  public mergeStopGroup(stopGroupToMergeGpuid: string, intoStopGroupGpuid: string): StopGroup {} 
+
+  /**
+   * @description Merge two stopClusters. It Means moving all stopGroup of a stopCluster into another.
+   * @param stopClusterToMergeGpuid Ground place unique identifier of the stopCluster to merge.
+   * @param intoStopClusterGpuid Ground place unique identifier of the stopCluster to be merged.
+   * @returns {StopCluster}
+   */
+  // @ts-ignore
+  // WARNING: A stopGroup can belong to both stopCluster, in this case, just remove it from the first stopCluster.
+  public mergeStopCluster(stopClusterToMergeGpuid: string, intoStopClusterGpuid: string): StopCluster {}
+
+  /**
+   * @description Check if all the business rules are respected. 
+   * Returns true if everything ok, throw an error with all issues if not.
+   * @returns {Boolean|Error}
+   */
+  // @ts-ignore
+  private isValid(): boolean | Error {}
```

Comment:
du coup c'est plutôt "true" | error ;) ?

Reply:
"is" that return an Error object ? It seems weird. Not sure that's the best way.

Reply:
We can only return **boolean** for `isValid() `, and only throw an **Error** if `!isValid()` ?

Reply:
Hmm, actually this function should return return all issues related to the error, so the naming is not revelent. We should discuss about it but I think that rename this by `checkValidity()` or synonym is appropriate in this situation. And also keep the Error object

Reply:
👍 for checkValidity